File: Task3-WebClient/src/services/UserService.js
import apiInstance from './ApiService';

export const getAllUsers = async () => {
    try {
        const response = await apiInstance.get('/api/user');
        return response.data;
    } catch (error) {
        console.error('Error fetching users:', error.response ? error.response.data : error.message);
        throw error;
    }
};

export const deleteUser = async (userId) => {
    try {
        await apiInstance.delete(`/api/user/delete/${userId}`);
    } catch (error) {
        console.error('Error deleting user:', error.response ? error.response.data : error.message);
        throw error;
    }
};

export const updateUser = async (userId, userData) => {
    try {
        if (!userId) throw new Error("User ID is undefined");
        const response = await apiInstance.put(`/api/user/edit/${userId}`, userData);
        console.log("User updated successfully:", response.data);
        return response.data;
    } catch (error) {
        console.error('Error updating user:', error.response ? error.response.data : error.message);
        throw error;
    }
};

export const createUser = async (userData) => {
    try {
        const response = await apiInstance.post('/api/user', userData);
        return response.data;
    } catch (error) {
        console.error('Error creating user:', error.response ? error.response.data : error.message);
        throw error;
    }
};

export const getUserByEmail = async (email) => {
    try {
      const response = await apiInstance.get(`/api/user/byEmail/${email}`);
      return response.data;
    } catch (error) {
      console.error('Error fetching user by email:', error);
      throw error;
    }
  };
  
  export const getManagerByEmail = async (email) => {
    try {
      const response = await apiInstance.get(`/api/manager/byEmail/${email}`);
      return response.data;
    } catch (error) {
      console.error('Error fetching manager by email:', error);
      throw error;
    }
  };

File: Task4-MobileClient/coworkings-mobile/coworkings-mobile/Services/UserService.cs
// Services/UserService.cs
using coworkings_mobile.Models;
using Newtonsoft.Json;
using System.Text;

namespace coworkings_mobile.Services
{
    public class UserService
    {
        private readonly HttpClient _httpClient;
        private readonly string _apiBaseUrl;

        public UserService()
        {
            _httpClient = new HttpClient();
            _apiBaseUrl = "http://though-forget.gl.at.ply.gg:9088"; 
        }

        public async Task<List<User>> GetAllUsers()
        {
            var response = await _httpClient.GetAsync($"{_apiBaseUrl}/api/user");
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                var users = JsonConvert.DeserializeObject<List<User>>(responseContent);
                return users;
            }
            else
            {
                return null;
            }
        }

        public async Task<bool> DeleteUser(int userId)
        {
            var response = await _httpClient.DeleteAsync($"{_apiBaseUrl}/api/user/delete/{userId}");

            return response.IsSuccessStatusCode;
        }

        public async Task<bool> UpdateUser(int userId, User userData)
        {
            var json = JsonConvert.SerializeObject(userData);
            var content = new StringContent(json, System.Text.Encoding.UTF8, "application/json");
            var response = await _httpClient.PutAsync($"{_apiBaseUrl}/api/user/edit/{userId}", content);

            if (response.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public async Task<User> CreateUser(User userData, string password)
        {

            var userWithPassword = new
            {
                password = password,
                email = userData.Email,
                fullName = userData.FullName
            };

            var json = JsonConvert.SerializeObject(userWithPassword);
            var content = new StringContent(json, Encoding.UTF8, "application/json");

            var response = await _httpClient.PostAsync($"{_apiBaseUrl}/api/user", content);
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                var createdUser = JsonConvert.DeserializeObject<User>(responseContent);
                return createdUser;
            }
            else
            {
                return null;
            }
        }
    }
}